src/main/java/chatbot/ChatBot.java

```diff
@@ -0,0 +1,117 @@
+package chatbot;
+
+import chatbot.exceptions.ChatBotException;
+import chatbot.exceptions.FilePermissionException;
+import chatbot.exceptions.IllegalCommandException;
+import chatbot.exceptions.LocalFileException;
+import chatbot.tasks.Task;
+
+/**
+ * Main ChatBot class which instantiates a ChatBot object that coordinates other components.
+ */
+public class ChatBot {
+    static final String NAME = "4F5DA2";
+    static final String LOCAL_DIRECTORY_PATH = "./data";
+    static final String LOCAL_FILE_PATH = LOCAL_DIRECTORY_PATH + "/chatbot.txt";
+    private final Ui ui;
+    private final Storage storage;
+    private TaskList tasks;
+    private boolean isExit = false; // Whether the user has indicated to exit the program.
+    private boolean noLocalFileAccess = false; // Whether the data file cannot be accessed due to permission issue.
+
+    /**
+     * Constructor to instantiate a new ChatBot object.
+     * @param directoryPath String path to the data file's directory
+     * @param filePath String path to the data file
+     */
+    public ChatBot(String directoryPath, String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(directoryPath, filePath);
+        try {
+            this.tasks = new TaskList(storage.readData());
+        } catch (LocalFileException e) {
+            this.ui.showLoadingError(e);
+            this.tasks = new TaskList();
+            if (e instanceof FilePermissionException) {
+                this.noLocalFileAccess = true;
+            }
+        }
+    }
+
+    private void run() {
+        this.ui.greet();
+        while (!isExit) {
+            try {
+                String command = this.ui.nextCommand();
+                this.ui.showLine();
+                this.handleCommand(command);
+            } catch (ChatBotException e) {
+                this.ui.output(e.toString());
+            }
+        }
+    }
+
+    private void handleCommand(String command) throws ChatBotException {
+        String[] words = command.split(" ");
+        String firstWord = words[0];
+        switch (firstWord) {
```

Comment:
Consider abstracting away the logic from the switch to make it more readable.

Reply:
Yep, better abstraction will be implemented in the future when I have time